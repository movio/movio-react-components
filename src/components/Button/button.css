@import 'variables.css';

:root {
  --spacing: 10px 10px;
  --bg-color: var(--color-primary);
}

.button-wrapper {
  border: 0;
  outline: 0;
}

.button {
  border: 0;
  outline: 0;
  padding: 1rem 1.5rem;
  color: white;
  background-color: var(--bg-color);
  transition: 0.2s background-color ease-in-out;

  @nest &:hover {
    cursor: pointer;
    background-color: color(var(--bg-color) b(+ 5%));
  }
}
