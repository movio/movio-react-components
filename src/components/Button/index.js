// @flow
import React from 'react';
import classnames from 'classnames';

import styles from './button.css';

type ClickHandler = (args?: any) => mixed;

type Props = {
  className?: string,
  onClick: ClickHandler,
  disabled?: boolean,
  children?: ReactChildren,
};

const Button = ({
  className = '',
  onClick,
  disabled = false,
  children,
  ...props }: Props) =>
    <button
      className={classnames(className, styles.button, { [styles.disabled]: disabled })}
      onClick={onClick}
      disabled={disabled === true}
      {...props}
    >
      {children}
    </button>;

export default Button;

export {
  styles,
};

