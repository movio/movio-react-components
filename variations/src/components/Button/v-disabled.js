module.exports = {
  "name": "Disabled",
  "props": {
    "className": "disabledButton",
    "innerClassName": "",
    "onClick": function () {
	    console.log('Run'); // eslint-disable-line
	  },
    "disabled": true,
    "children": "Disabled"
  }
};