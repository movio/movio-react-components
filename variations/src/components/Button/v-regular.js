module.exports = {
  "name": "Regular",
  "props": {
    "className": "btn",
    "onClick": function () {
	    console.log('Run'); // eslint-disable-line
	  },
    "children": "Button",
    "disabled": false
  }
};