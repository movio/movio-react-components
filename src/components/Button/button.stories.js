import React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import Button, { styles } from './index';

import scaffoldingStyles from '../../../.storybook/scaffolding.css';

const clickHandler = action('click');

storiesOf('Button', module)
  .add('primary', () => <Button onClick={clickHandler}>Primary</Button>)
  .add('secondary', () => (
    <Button onClick={clickHandler} secondary={true}>
      Secondary
    </Button>
  ))
  .add('disabled', () => (
    <Button onClick={clickHandler} disabled={true}>Disabled</Button>
  ))
  .add('loading', () => (
    <Button onClick={clickHandler} loading={true}>Loading</Button>
  ))
  .add('custom class', () => (
    <Button onClick={clickHandler} className={styles.loading}>Custom</Button>
  ))
  .add('all', () => (
    <div className={scaffoldingStyles.horizontalContainer}>
      <Button onClick={clickHandler}>Primary</Button>
      <Button onClick={clickHandler} secondary={true}>
        Secondary
      </Button>
      <Button onClick={clickHandler} disabled={true}>Disabled</Button>
      <Button onClick={clickHandler} loading={true}>Loading</Button>
      <Button onClick={clickHandler} className={styles.loading}>Custom</Button>
    </div>
  ));
