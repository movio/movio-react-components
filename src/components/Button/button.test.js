import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import Button, { styles } from './index';

const noop = () => {};

describe('<Button />', () => {
  it('should correctly render children, when text', () => {
    const component = <Button onClick={noop}>This is a button</Button>;
    const wrapper = shallow(component);
    expect(wrapper.text()).toEqual('This is a button');
    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot('button-children-text');
  });

  it('should append a className to the default, if provided', () => {
    const component = <Button className="test-class" onClick={noop}>Movio Button</Button>;
    const wrapper = shallow(component);
    expect(wrapper.prop('className')).toEqual(`test-class ${styles.button}`);
    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot('button-custom-classname');
  });

  it('should use the default className from the stylesheet, if not provided', () => {
    const component = <Button onClick={noop}>Movio Button</Button>;
    const wrapper = shallow(component);
    expect(wrapper.prop('className')).toEqual(styles.button);
    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot('button-default-classname');
  });

  it('should add a disabled class to the button, if disabled', () => {
    const component = <Button onClick={noop} disabled={true}>Disabled Button</Button>;
    const wrapper = shallow(component);
    expect(wrapper.prop('className')).toEqual(`${styles.button} ${styles.disabled}`);
    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot('button-disabled');
  });

  it('should add an onClick handler to the button', () => {
    const wrapper = shallow(<Button onClick={noop}>Button</Button>);
    expect(wrapper.prop('onClick')).toEqual(noop);
  });

  it('should trigger a click when onClick is fired', () => {
    const btnSpy = jest.fn();
    const wrapper = shallow(<Button onClick={btnSpy}>Button</Button>);
    wrapper.find('button').simulate('click');
    expect(btnSpy.mock.calls).toHaveLength(1);
  });
});
