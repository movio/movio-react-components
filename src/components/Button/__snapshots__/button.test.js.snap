// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button-children-text 1`] = `
<button
  className="button"
  disabled={false}
  onClick={[Function]}
>
  This is a button
</button>
`;

exports[`button-custom-classname 1`] = `
<button
  className="test-class button"
  disabled={false}
  onClick={[Function]}
>
  Movio Button
</button>
`;

exports[`button-default-classname 1`] = `
<button
  className="button"
  disabled={false}
  onClick={[Function]}
>
  Movio Button
</button>
`;

exports[`button-disabled 1`] = `
<button
  className="button disabled"
  disabled={true}
  onClick={[Function]}
>
  Disabled Button
</button>
`;
